### PR TITLE
GSGGR-156 implement authentication for the frontend-authenticated user

### PR DESCRIPTION
**Pull request by lanseg**
Flow:

1. Frontend authenticates in Zitadel, receives an access token and sends it to backend.
2. Backend uses service key and gets user information from the OIDC provider for given access token.
3. If key validation fails then return error. If everything is ok, then user is created and updated.
4. Send a refresh and access tokens to the frontend.

Configuration requires an extra account on the Zitadel side (of a General API type)

### CODE REVIEW

**Review by hbollon — APPROVED** [review]
LGTM :+1:

File: oidc.py [hunk 1]
```diff
@@ -1,26 +1,112 @@
...
-def updateUser(user, claims):
+def _updateUser(user, claims):
     user.email = claims.get("email")
     user.first_name = claims.get("given_name")
     user.last_name = claims.get("family_name")
     return user
 
+
+def _read_private_key(keyfile):
+    with open(keyfile, "r") as f:
```
Comment (danduk82) [inline oidc.py:29]: maybe add a `try` and catch the case where the env variable `OIDC_PRIVATE_KEYFILE` is not set, with an explicit error

**Review by danduk82 — APPROVED** [review]
ok for me as well, one little comment but lgtm also in the current state

File: oidc.py (hunk 1 above)
Comment (lanseg) [reply oidc.py:29]: Now settings throws an exception if the OIDC is enabled, but misses some configuration parameters

**Pull request merged**